CMakeLists.txt

```diff
@@ -287,27 +287,31 @@ if(BUILD_EXAMPLES AND NOT WIN32)
     target_compile_definitions(hello_module PRIVATE ${qjs_defines})
     target_link_libraries(hello_module ${qjs_libs})
 
-    if(NOT WIN32)
-        add_library(fib MODULE examples/fib.c)
-        set_target_properties(fib PROPERTIES
-            PREFIX ""
-            C_VISIBILITY_PRESET default
-        )
-        target_compile_definitions(fib PRIVATE JS_SHARED_LIBRARY)
-        if(APPLE)
-            target_link_options(fib PRIVATE -undefined dynamic_lookup)
-        endif()
+    add_library(fib MODULE examples/fib.c)
+    set_target_properties(fib PROPERTIES
+        PREFIX ""
+        C_VISIBILITY_PRESET default
+    )
+    target_compile_definitions(fib PRIVATE JS_SHARED_LIBRARY)
+    if(WIN32)
+        target_link_libraries(fib ${qjs_libs})
```

Comment:
For my curiosity, what library or libraries in particular are necessary? You normally only use qjs_libs for binaries, not shared libraries.

Reply:
I looked at this at one point. The problem seems to be weak symbol linking is not a thing on windows (?) or I didn't know the right flags!

Reply:
Is there something (like functions from math.h) leaking out of quickjs.h that necessitates that? Or is it something else?

Reply:
IIrC it's the quickjs functions themselves.

Reply:
Okay, that makes sense, but wouldn't it be best to link against just quickjs.lib?

Reply:
Fair point!